{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "import anndata as ad\n",
        "import scanpy as sc\n",
        "import pandas as pd\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "# Load podd et al data with log2(TPM+1) transformed \n",
        "podd_et_al = pd.read_csv(\"/Users/z5155527/Desktop/Benchmark-2025-Sep/phase_1_datasets/bulk_RNA-seq/podd_et_al/GSE158403_raw_counts_GRCh38.p13_NCBI.tsv\", sep = '\\t')\n",
        "podd_et_al.index = podd_et_al[\"GeneID\"].tolist()\n",
        "podd_et_al = podd_et_al.drop(columns=\"GeneID\").T\n",
        "podd_et_al\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "# read clinical data of podd_et_al\n",
        "podd_et_al_clinical = pd.read_excel(\"/Users/z5155527/Desktop/Benchmark-2025-Sep/phase_1_datasets/bulk_RNA-seq/podd_et_al/podd_et_al_clinical.xlsx\")\n",
        "podd_et_al_clinical\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "# map sample list of clinical data\n",
        "clinical_sample_list = podd_et_al_clinical[\"Sample.ID\"].tolist()\n",
        "sample_list_mapping = pd.read_excel(\"/Users/z5155527/Desktop/Benchmark-2025-Sep/phase_1_datasets/bulk_RNA-seq/podd_et_al/podd_et_al_sample_mapping.xlsx\")\n",
        "# Check the overlap between clinical_sample_list and sample_list_mapping\n",
        "# Assuming the mapping file has columns \"Sample_title\" and \"Sample_geo_accession\"\n",
        "mapped_sample_list = sample_list_mapping[\"Sample_title\"].tolist()\n",
        "\n",
        "# Convert to sets for comparison\n",
        "clinical_sample_set = set(clinical_sample_list)\n",
        "mapped_sample_set = set(mapped_sample_list)\n",
        "\n",
        "# Find intersection and differences\n",
        "overlap_samples = clinical_sample_set & mapped_sample_set\n",
        "only_in_clinical = clinical_sample_set - mapped_sample_set\n",
        "only_in_mapping = mapped_sample_set - clinical_sample_set\n",
        "\n",
        "print(f\"Number of overlapping samples: {len(overlap_samples)}\")\n",
        "print(f\"Samples only in clinical: {only_in_clinical}\")\n",
        "print(f\"Samples only in mapping: {only_in_mapping}\")\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "# map the sample name to gene expression data\n",
        "# Map the sample names in podd_et_al (currently the index) to GEO accession using the sample_list_mapping\n",
        "# Sample_list_mapping has columns: Sample_title (matching the clinical data and podd_et_al index) and Sample_geo_accession (matching expression data)\n",
        "\n",
        "# Set Sample_title as index for easier lookup\n",
        "sample_list_mapping_indexed = sample_list_mapping.set_index(\"Sample_title\")\n",
        "\n",
        "# Map Sample_geo_accession to expression data index using the Sample_title\n",
        "# If podd_et_al.index are GEO accessions (e.g. GSM...), then we want to reverse-map:\n",
        "# Otherwise, we create a mapping dict: Sample_title => Sample_geo_accession\n",
        "sample_title_to_geo = sample_list_mapping.set_index(\"Sample_title\")[\"Sample_geo_accession\"].to_dict()\n",
        "\n",
        "# Now, try to map podd_et_al's index (GEO accession) to sample title (using the reverse mapping)\n",
        "geo_to_sample_title = sample_list_mapping.set_index(\"Sample_geo_accession\")[\"Sample_title\"].to_dict()\n",
        "\n",
        "# Add column to podd_et_al for sample_title (if present in mapping)\n",
        "podd_et_al[\"sample_title\"] = podd_et_al.index.map(geo_to_sample_title)\n",
        "\n",
        "# If you want to save the mapping result, or check alignment:\n",
        "print(podd_et_al[[\"sample_title\"]].head())\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "podd_et_al_tosave = podd_et_al.copy()\n",
        "podd_et_al_tosave.index = podd_et_al_tosave[\"sample_title\"]\n",
        "podd_et_al_tosave.drop(columns=[\"sample_title\"], inplace=True)\n",
        "podd_et_al_tosave\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "# now map the gene names through the table\n",
        "human_annotation_table = pd.read_csv(\"/Users/z5155527/Desktop/Benchmark-2025-Sep/phase_1_datasets/bulk_RNA-seq/sup/Human.GRCh38.p13.annot.tsv\", sep = '\\t')\n",
        "human_annotation_table\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "# map GeneID to columns of podd_et_al_tosave\n",
        "# Create a mapping from GeneID (as string) to Symbol\n",
        "geneid_to_symbol = dict(zip(human_annotation_table[\"GeneID\"], human_annotation_table[\"Symbol\"]))\n",
        "# Optional: check a few columns after renaming\n",
        "#podd_et_al_tosave\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "\n"
      ]
    }
  ],
  "metadata": {
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 2
}
